Sources/NetworkService/NetworkError.swift

```diff
@@ -0,0 +1,37 @@
+import Foundation
+
+/// Contains network errors that may occur, while the application is running.
+public enum NetworkError: Error, CaseIterable {
+    /// Indicates that the Internet connection has been lost.
+    case lostConnection
```

Comment:
Do we need to add a case about missing data, when the response was successful, but no data comes back?